inginious/frontend/templates/course_admin/edit_tabs/tags.html

```diff
@@ -0,0 +1,148 @@
+        $for t in list_index:
+            $ i += 1
+            <tr>
+                <td id="A-$i" style="padding:4px">$t.get_id()</td>
+                <td id="B-$i" style="padding:4px">$t.get_name()</td>
+                <td id="C-$i" style="padding:4px">$t.get_description()</td>
+                <td id="D-$i" style="padding:4px">$t.is_visible_for_student()</td>
+                <td id="E-$i" style="padding:4px" data-type="$t.get_type()" >$t.get_type_as_str()</td>
+                <td><button type="button" onclick="add_tag_line($i)" class="btn btn-primary btn-xs"><i class="fa fa-arrow-down" aria-hidden="true"></i></button></td>
+            </tr>
+    </tbody>
+</table>
+
+</div>
+</div>
+<!-- Menu Import tag END-->
+    
+<script>
+function change_id_field(type, line) {
+    if(type.value == 2){    
+        $$("#id_"+line).prop('disabled', true);
+    }else{
+        $$("#id_"+line).prop('disabled', false);
+    }
+}
+function expand(elem){
+    elem.rows = 5;
+}
+function expand_not(elem){
+    elem.rows = 1;
+}
+function add_tag_line(line=-1) {    
+    var id = $$('#A-'+line).text();
+    var name = $$('#B-'+line).text();
+    var descr = $$('#C-'+line).text();
+    var visibility = $$('#D-'+line).text();
+    var type = $$('#E-'+line).attr("data-type");
+
+    if (visibility == "True"){
+        visibility = "checked";
+    }
+    var checked0 = '';
+    var checked1 = '';
+    var checked2 = '';
+    var disable_id = '';
+    if(type == "2"){
+        checked2 = 'selected="selected"';
+        disable_id = ' disabled '
+    }
+    else if(type == "1"){
+        checked1 = 'selected="selected"';
+    }
+    else{
+        checked0 = 'selected="selected"';
+    }
+    var new_id = 1 + parseInt($$('#table tr:last').attr('id'));
+    if (isNaN(new_id))
+        new_id = 0
+    var new_row = '\
+        <tr id='+ new_id +'> \
+            <td><input type="text" '+disable_id+'id="id_'+ new_id +'" class="form-control" name="tags['+ new_id +'][id]" value="'+id+'"/></td> \
+            <td><input type="text" class="form-control" name="tags['+ new_id +'][name]" value="'+name+'"/></td> \
+            <td><textarea class="form-control" name="tags['+ new_id +'][description]" onfocus="expand(this);" onblur="expand_not(this);" rows="1" style="resize:none">'+descr+'</textarea></td> \
+            <td><input type="checkbox" class="form-control" name="tags['+ new_id +'][visible]" '+visibility+'/></td> \
+            <td> \
+                <select class="form-control" name="tags['+ new_id +'][type]" onchange="change_id_field(this, '+ new_id +')"> \
+                    <option value="0" '+checked0+'>$:_("Common")</option> \
+                    <option value="1" '+checked1+'>$:_("Antitag")</option> \
+                    <option value="2" '+checked2+'>$:_("Organisation")</option> \
+                </select> \
+            </td> \
+        </tr>'
+    $$('#table').find('tbody').append(new_row);
+}
+</script>
```

Comment:
I'd rather have the script part in the ``studio.js``file. However, I'd put the big html snippet directly into the html code as a hidden template and set the appropriate field in JS after copying the object.